res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+//    - Sync
+//    - Volume fader
+//    - EQs
+//    - PFL
+//    - Pad Selectors
+//    - Loop PADs
+//    - Roll PADs
+//    - Beat jump PADs
+//    - Tone key PADs
+//    - Slicer
+//    - Loop pot
+//    - In and Out loop
+//    - Load button
+//    - Vinyl
+//    - Slip
+//    - Quant
+//    - Pitch fader
+//    - Jog wheels (Using the new JogWheelBasic component!
+//      - Also probably fixed the shift behavior not working properly
+//
+//  * Added option so the browser knob can behave with out of focus window
+//
+// * Version 1.5c (Summer 2023)
+// * Forum: https://mixxx.discourse.group/t/hercules-djcontrol-inpulse-500/19739
+// * Wiki: https://mixxx.org/wiki/doku.php/hercules_djcontrol_inpulse_500
+//
+//  Version 1.0c:
+//	* Hot Cue: implementation of the Color API (Work in progress)
+//		- Assigned color directly to pad (XML)
+//	* Added DECK LED number - On when playing
+//  * Moved Beatjump to Pad mode 3 (Slicer)
+//	* Set different color for upper (Sampler 1-4) and lower (Sampler 5-8) sampler pads
+//
+//  Version 1.0 - Based upon Inpulse 300 v1.2 (official)
+//
+// TO DO:
+//  * Browser knob has a ton of colors to do things!
+//  * Vinyl + SHIFT led should reflect brake status
+//  * Quant + SHIFT led should reflect key lock status
+//  * Add beat jump + SHIFT jumps
+//
+// ****************************************************************************
+var DJCi500 = {};
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// If you are spinning your set list and you have your Mixxx window out
+// of focus and you want to be able to use the browser knob to traverse
+// the current crate or playlist, set to true. Especially useful to spin
+// when using Twitch, VRChat or Second Life
+DJCi500.browserOffFocusMode = false;
+
+// Set initial state for vinyl mode button
+DJCi500.initialVinylMode = true;
+
+// Colors
+DJCi500.PadColorMapper = new ColorMapper({
+  0xFF0000: 0x60,
+  0xFFFF00: 0x7C,
+  0x00FF00: 0x1C,
+  0x00FFFF: 0x1F,
+  0x0000FF: 0x03,
+  0xFF00FF: 0x42,
+  0xFF88FF: 0x63,
+  0xFFFFFF: 0x7F,
+  0x000088: 0x02,
+  0x008800: 0x10,
+  0x008888: 0x12,
+  0x228800: 0x30,
+  0x880000: 0x40,
+  0x882200: 0x4C,
+  0x888800: 0x50,
+  0x888888: 0x52,
+  0x88FF00: 0x5C,
+  0xFF8800: 0x74,
+});
+
+// Constants
+
+DJCi500.EFFECT_ONLY_MODE = 1;
+DJCi500.FILTER_AND_EFFECT_MODE = 2;
+
+// For key shift pads and beat jump pads
+const pairColorsOn = [0x1F, 0x1F, 0x03, 0x03, 0x74, 0x74, 0x60, 0x60];
+const pairColorsOff = [0x12, 0x12, 0x02, 0x02, 0x4C, 0x4C, 0x40, 0x40];
+
+
+///////////////////////////////////////////////////////////////
+//                          SLICER                           //
+///////////////////////////////////////////////////////////////
+DJCi500.selectedSlicerDomain = [8, 8, 8, 8]; //length of the Slicer domain
+//PioneerDDJSX.slicerDomains = [8, 16, 32, 64];

```

Comment:
The second line seems to be a copy-paste leftover.

```suggestion
DJCi500.selectedSlicerDomain = [8, 8, 8, 8]; // Length of the Slicer domain
//PioneerDDJSX.slicerDomains = [8, 16, 32, 64];
```